### PR TITLE
feat: accept legacy penalty reference numbers

### PR DESCRIPTION
**Pull request by akaJSaunders**
### JIRA link

https://companieshouse.atlassian.net/browse/LFA-1653

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are commited to keeping commit history clean, consistent and linear. To achive this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/PenaltyReferenceSanitizer.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { sanitizeCompany } from 'app/utils/CompanyNumberSanitizer';
+
+export const sanitizePenalty = (penaltyReference: string): string => {
+
+    penaltyReference = penaltyReference.toUpperCase();
+    const prefix = penaltyReference.substring(0,6);
```
Comment (n-r-b) [inline src/utils/PenaltyReferenceSanitizer.ts:6]: isn't the prefix 6 characters, therefore this would be `....substring(0,5);` ?

Comment (n-r-b) [reply src/utils/PenaltyReferenceSanitizer.ts:6]: Can you please add the type

File: src/utils/PenaltyReferenceSanitizer.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    const companyNumber = sanitizeCompany(penaltyReference.substring(6, penaltyReference.length));
```
Comment (n-r-b) [inline src/utils/PenaltyReferenceSanitizer.ts:7]: Can you please add the type

File: test/models/PenaltyEntrySchema.test.ts [hunk 3]
```diff
@@ -131,6 +131,42 @@ describe('Penalty Details Schema Validation', () => {
...
+                const result = validator.validate(createModelWithPenaltyReference('pen1A/sc123123'));
+                expect(result).to.deep.equal({errors: []});
+            });
+
+            it('should accept legacy penalty references with leading zeros', () => {
+                const result = validator.validate(createModelWithPenaltyReference('pen1A/sc123'));
+                expect(result).to.deep.equal({errors: []});
+            });
+
+            it('should accept legacy penalty references with leading zeros', () => {
```
Comment (n-r-b) [inline test/models/PenaltyEntrySchema.test.ts:150]: looks like this is a duplicate test

File: test/models/PenaltyEntrySchema.test.ts [hunk 4]
```diff
@@ -131,6 +131,42 @@ describe('Penalty Details Schema Validation', () => {
...
+                expect(result).to.deep.equal({errors: []});
+            });
+
+            it('should accept lower case legacy penalty references', () => {
+                const result = validator.validate(createModelWithPenaltyReference('pen1A/sc123123'));
+                expect(result).to.deep.equal({errors: []});
+            });
+
+            it('should accept legacy penalty references with leading zeros', () => {
+                const result = validator.validate(createModelWithPenaltyReference('pen1A/sc123'));
```
Comment (n-r-b) [inline test/models/PenaltyEntrySchema.test.ts:146]: are you missing the leading zeros?

File: test/models/PenaltyEntrySchema.test.ts [hunk 5]
```diff
@@ -131,6 +131,42 @@ describe('Penalty Details Schema Validation', () => {
                 const result = validator.validate(createModelWithPenaltyReference('z12345678'));
                 expect(result).to.deep.equal({errors: []});
             });
+
+            it('should accept legacy penalty references', () => {
```
Comment (n-r-b) [inline test/models/PenaltyEntrySchema.test.ts:135]: How about a test for example on ticket: `PEN1A/12345678`?

File: src/utils/PenaltyReferenceSanitizer.ts (hunk 1 above)
Comment (akaJSaunders) [reply src/utils/PenaltyReferenceSanitizer.ts:6]: last number isnt inclusive

File: test/models/PenaltyEntrySchema.test.ts (hunk 4 above)
Comment (akaJSaunders) [reply test/models/PenaltyEntrySchema.test.ts:146]: SC123 is a valid company number, it will be saved on our system as SC000123

Comment (akaJSaunders) [reply test/models/PenaltyEntrySchema.test.ts:146]: ill change the name

**Pull request merged**